### PR TITLE
Check whether binstubs is installed expectedly

### PR DESCRIPTION
**Pull request by kenhys**
bundle binstubs --all --path ... should be exist on
/opt/td-agent/bin

### CODE REVIEW

**Comment by kenhys** [comment]
Just waiting for CI.

**Comment by kenhys** [comment]
missing command issue had fixed https://github.com/fluent-plugins-nursery/td-agent-builder/pull/293, but no mechanism to check it yet with systematically, 
assigned to milestone 4.1.1.

**Comment by kenhys** [comment]
It needs still some tweak for aarch64.

**Comment by cosmo0920** [comment]
(nits....? As outsiders, I didn't find this PR works as expected: https://github.com/fluent-plugins-nursery/td-agent-builder/pull/294/checks?check_run_id=2396163796#step:11:631 Is it intended behavior....??)

**Comment by kenhys** [comment]
> Is it intended behavior....??)

Oops, not intended, I'll fix it.

**Comment by kenhys** [comment]
It has passed on some platforms. waiting result for aarch64 and windows.


```
2021-04-22T02:48:14.4707182Z OK: /opt/td-agent/bin/fluent-plugin-generate exists
2021-04-22T02:48:14.4708021Z OK: /opt/td-agent/bin/fluent-ca-generate exists
2021-04-22T02:48:14.4708696Z OK: /opt/td-agent/bin/td exists
2021-04-22T02:48:14.4709257Z OK: /opt/td-agent/bin/rake exists
2021-04-22T02:48:14.4709897Z OK: /opt/td-agent/bin/fluent-post exists
2021-04-22T02:48:14.4710578Z OK: /opt/td-agent/bin/fluent-debug exists
2021-04-22T02:48:14.4711219Z OK: /opt/td-agent/bin/nokogiri exists
2021-04-22T02:48:14.4712020Z OK: /opt/td-agent/bin/elastic_ruby_console exists
2021-04-22T02:48:14.4712671Z OK: /opt/td-agent/bin/bundle exists
2021-04-22T02:48:14.4713268Z OK: /opt/td-agent/bin/racc exists
2021-04-22T02:48:14.4714050Z OK: /opt/td-agent/bin/fluent-diagtool exists
2021-04-22T02:48:14.4714739Z OK: /opt/td-agent/bin/fluent-cat exists
2021-04-22T02:48:14.4715570Z OK: /opt/td-agent/bin/fluent-plugin-config-format exists
2021-04-22T02:48:14.4716391Z OK: /opt/td-agent/bin/fluent-ctl exists
2021-04-22T02:48:14.4717069Z OK: /opt/td-agent/bin/fluent-cap-ctl exists
2021-04-22T02:48:14.4717734Z OK: /opt/td-agent/bin/fluentd exists
2021-04-22T02:48:14.4718456Z OK: /opt/td-agent/bin/fluent-binlog-reader exists
2021-04-22T02:48:14.4719199Z OK: /opt/td-agent/bin/httpclient exists
2021-04-22T02:48:14.4719842Z OK: /opt/td-agent/bin/fluent-gem exists
2021-04-22T02:48:14.4720479Z OK: /opt/td-agent/bin/bundler exists
2021-04-22T02:48:15.5995162Z Post job cleanup.
2021-04-22T02:48:15.6875537Z [command]
```

**Comment by kenhys** [comment]
aarch64 specific issue was fixed.

File: td-agent/yum/binstubs-test.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        else
+            DISTRIBUTION_VERSION=${version}
+        fi
+        ;;
+    esac
+    ;;
+esac
+
+repositories_dir=/fluentd/td-agent/yum/repositories
+ARCH=$(uname -m)
```
Comment (cosmo0920) [inline td-agent/yum/binstubs-test.sh:51]: For consistency, how about obtain architechtures from `rpm --eval "%{_arch}"`?
We retrieve architecture information with dpkg which is package management commant on apt environments.
We should also retrieve it from package management command which is `rpm`.
WDYT?

Comment (kenhys) [reply td-agent/yum/binstubs-test.sh:51]: Thanks!
it is better to use that way. I'll fix it

Comment (cosmo0920) [reply td-agent/yum/binstubs-test.sh:51]: FYI: `uname -m` is almost always returning `x86_64`. This command returns docker container's host kernel architecture.

File: .github/workflows/yum-arm.yml [hunk 2]
```diff
@@ -44,3 +44,13 @@ jobs:
...
           td-agent/yum/pkgsize-test.sh ${{ matrix.rake-job }} aarch64
+      - name: Binstubs Test
+        run: |
+          mkdir -p .bundle
+          docker run \
+          --rm \
+          --tty \
+          --volume ${PWD}:/fluentd:ro \
+          --env CENTOS_STREAM=false \
+          ${{ matrix.test-docker-image }} \
```
Comment (cosmo0920) [inline .github/workflows/yum-arm.yml:55]: This workflow still uses x86_64 images for centos and amazonlinux 2.
Should we use `arm64v8/` prefixed images like as apt-arm workflow?

WDYT?

Comment (kenhys) [reply .github/workflows/yum-arm.yml]: Oh, I've overlooked it, thanks.

Comment (kenhys) [reply .github/workflows/yum-arm.yml]: resolved.

**Comment by ashie** [comment]
What the status of this PR? Already completed?
I'll this merge this after merging #294 
I'll check the td and elastic_ruby_console commands manually for td-agent 4.1.1

**Comment by kenhys** [comment]
The rest of the issues which are pointed out by @cosmo0920 has been fixed now.

**Pull request merged**